### DISCUSSION TITLE
How to change DisableURIPathEscaping ?

### DISCUSSION THREAD

**Discussion by leavest** [aws/aws-sdk-go-v2#2804] [General]
I GetObject use Chinese key but get error  "The request signature we calculated does not match the signature you provided.".
I change DisableURIPathEscaping to true in source and get success.

aws source source code
new client with default DisableURIPathEscaping=true

https://github.com/aws/aws-sdk-go-v2/blob/e211ac0d542b6529cb7f81c05d230e5a2a6f91d6/service/s3/api_client.go#L631

but in Signer.SignHTTP

https://github.com/aws/aws-sdk-go-v2/blob/e211ac0d542b6529cb7f81c05d230e5a2a6f91d6/aws/signer/v4/v4.go#L276

DisableURIPathEscaping  change to false 
optFns has one member V4SignerAdapter.SignRequest

https://github.com/aws/aws-sdk-go-v2/blob/e211ac0d542b6529cb7f81c05d230e5a2a6f91d6/internal/auth/smithy/v4signer_adapter.go#L46

after this line DisableURIPathEscaping change to false

How to change DisableURIPathEscaping in a new request ?
my code 

```
        client := s3.New(s3.Options{
		EndpointResolverV2: &ResolverV2{URL: endpoint},
		Credentials: aws.CredentialsProviderFunc(func(ctx context.Context) (aws.Credentials, error) {
			return aws.Credentials{
				AccessKeyID:     AK,
				SecretAccessKey: SK,
			}, nil
		}),
		RetryMaxAttempts: 1,
	})
        input := s3.GetObjectInput{
		Bucket: &bucket,
		Key:    &req.Prefix,
	}
	output, err := client.GetObject(c, &input)
```

**Comment by autowp** [comment]
Thinks i found solution here https://github.com/aws/aws-sdk-go-v2/issues/3349
